OpenRA.Game/Server/Server.cs

```diff
@@ -746,6 +771,46 @@ public void DispatchOrdersToClients(Connection conn, int frame, byte[] data)
 			RecordOrder(frame, data, from);
 		}
 
+		void DispatchOrdersToClient(Connection c, int client, int frame, byte[] data, bool buffer = false)
+		{
+			DispatchFrameToClient(c, CreateFrame(client, frame, data), buffer);
+		}
+
+		void DispatchFrameToClient(Connection c, byte[] frameData, bool buffer = false)
+		{
+			if (buffer)
+				c.SendBufferedData(frameData);
+			else
+				c.SendData(frameData);
+		}
+
+		public void DispatchBufferedOrdersToOtherClients(int fromClient, List<byte[]> allData)
+		{
+			var ms = new MemoryStream(allData.Select(d => d.Length).Sum());
+			foreach (var data in allData)
+				ms.WriteArray(data);
+
+			var conn = Conns.FirstOrDefault(c => c.PlayerIndex == fromClient);
```

Comment:
```suggestion
			var conn = Conns.First(c => c.PlayerIndex == fromClient);
```
If this doesn't find anything it will NRE in `DispatchOrdersToClients`.

Reply:
It prolly should be `Single(...)` for explicitness.